### PR TITLE
Fix java compiler warnings

### LINKED ISSUE #5590: Numerous compiler warnings
### Problem

When running gradle build we see warnings about potentially unsafe operations. We should have a clean build that's warning free so we don't become complacent in ignoring warnings.

```
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
```

### Solution

* Add `-Xlint:all` compiler args
* Fix any warnings during `compileJava`
* Fix low hanging fruit during `compileTestJava` but probably too many to fix all

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by steven-sheehy**
**Description**:

* Add compiler args to fail build for any compiler warnings
* Change `npm pack` to not print verbose output to stderr
* Fix java compiler warnings

**Related issue(s)**:

Fixes #5590

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-monitor/src/main/java/com/hedera/mirror/monitor/properties/ScenarioPropertiesAggregatorImpl.java [hunk 1]
```diff
@@ -47,9 +49,11 @@ public Map<String, Object> aggregateProperties(Map<String, String> properties) {
...
                 log.debug("Converting property {} into list {}", entry.getKey(), propertyName);
                 correctedProperties
-                        .merge(propertyName, Lists.newArrayList(entry.getValue()), (existingList, newList) -> {
-                            ((List) existingList).addAll((List) newList);
-                            return existingList;
+                        .merge(propertyName, Lists.newArrayList(entry.getValue()), (e, n) -> {
+                            if (e instanceof List<?> existingList && n instanceof Collection newList) {
+                                existingList.addAll(newList);
+                            }
+                            return e;
```
Comment (edwin-greene) [inline hedera-mirror-monitor/src/main/java/com/hedera/mirror/monitor/properties/ScenarioPropertiesAggregatorImpl.java:56]: Maybe there should be an else clause here. If a property isn't converted into a list correctly then a warning is logged.

File: hedera-mirror-monitor/src/main/java/com/hedera/mirror/monitor/properties/ScenarioPropertiesAggregatorImpl.java [hunk 2]
```diff
@@ -37,6 +38,7 @@ public class ScenarioPropertiesAggregatorImpl implements ScenarioPropertiesAggre
             .compile("(\\w+)\\.\\d+");
 
     @Override
+    @SuppressWarnings("unchecked")
```
Comment (edwin-greene) [inline hedera-mirror-monitor/src/main/java/com/hedera/mirror/monitor/properties/ScenarioPropertiesAggregatorImpl.java:41]: Seems `unchecked` is not needed. Sonar is complaining about `java:S3740`.

File: hedera-mirror-common/src/main/java/com/hedera/mirror/common/domain/transaction/RecordItem.java
Comment (edwin-greene) [inline hedera-mirror-common/src/main/java/com/hedera/mirror/common/domain/transaction/RecordItem.java:1]: Looks like line 183 can now be simplified: 
`public static class RecordItemBuilder {`

**Review by MarcKriguerAtHedera — APPROVED** [review]
LGTM -- nice!

**Review by edwin-greene — APPROVED** [review]
Looks good!

**Pull request merged**